File: examples/android-utils/src/main/java/com/gurunars/android_utils/example/storage/StringSerializer.java
package com.gurunars.android_utils.example.storage;


import android.support.annotation.Nullable;
import android.util.Base64;
import android.util.Base64InputStream;
import android.util.Base64OutputStream;

import java.io.ByteArrayInputStream;
import java.io.ByteArrayOutputStream;
import java.io.IOException;
import java.io.ObjectInputStream;
import java.io.ObjectOutputStream;
import java.io.Serializable;

class StringSerializer {

    /** Read the object from Base64 string. */
    @Nullable
    static <ObjType extends Serializable> ObjType fromString(@Nullable String serializedObject ) {
        if (serializedObject == null) {
            return null;
        }
        try {
            return (ObjType) new ObjectInputStream(
                new Base64InputStream(
                    new ByteArrayInputStream(
                        serializedObject.getBytes()), Base64.NO_PADDING | Base64.NO_WRAP)
            ).readObject();
        } catch (Exception e) {
            e.printStackTrace();
        }
        return null;
    }

    /** Write the object to a Base64 string. */
    @Nullable
    static String toString(@Nullable Serializable obj ) {
        if (obj == null) {
            return null;
        }
        try {
            ByteArrayOutputStream baos = new ByteArrayOutputStream();
            ObjectOutputStream oos = new ObjectOutputStream(
                    new Base64OutputStream(baos, Base64.NO_PADDING
                            | Base64.NO_WRAP));
            oos.writeObject(obj);
            oos.close();
            return baos.toString("UTF-8");
        } catch (IOException e) {
            e.printStackTrace();
        }
        return null;
    }

}

File: android-utils/src/main/kotlin/com/gurunars/storage/StringSerializer.kt
package com.gurunars.storage

import android.util.Base64
import android.util.Base64InputStream
import android.util.Base64OutputStream
import java.io.*
import java.lang.Exception


internal object StringSerializer {

    /** Read the object from Base64 string.  */
    fun <ObjType> fromString(serializedObject: String?): ObjType? {
        if (serializedObject == null) return null
        try {
            @Suppress("UNCHECKED_CAST")
            return ObjectInputStream(
                Base64InputStream(
                    ByteArrayInputStream(serializedObject.toByteArray()),
                    Base64.NO_PADDING or Base64.NO_WRAP)
            ).readObject() as ObjType
        } catch (e: Exception) {
            e.printStackTrace()
            return null
        }
    }

    /** Write the object to a Base64 string.  */
    fun toString(obj: Any?): String? {
        if (obj == null) return null
        try {
            val baos = ByteArrayOutputStream()
            val oos = ObjectOutputStream(
                Base64OutputStream(
                    baos,
                    Base64.NO_PADDING or Base64.NO_WRAP))
            oos.writeObject(obj)
            oos.close()
            return baos.toString("UTF-8")
        } catch (e: IOException) {
            e.printStackTrace()
            return null
        }
    }

}
